File: src/lib/api/codexReset.ts
import { invoke } from "@tauri-apps/api/core";

/** 全局重置预告（社区数据源 codex-reset.com 的公开 feed）。所有字段可缺省。 */
export interface CodexResetFeed {
  /** 上次已验证重置（epoch 秒）+ 原帖链接 */
  lastVerifiedAt: number | null;
  lastVerifiedUrl: string | null;
  /** 最新一条重置公告（epoch 秒 + 原帖链接 + 原文摘要） */
  announcementAt: number | null;
  announcementUrl: string | null;
  announcementSummary: string | null;
  /** 公告文本窄解析出的预计落地时间（epoch 秒）；解析不出为 null */
  landingAt: number | null;
  fetchedAt: number;
}

export const codexResetApi = {
  getFeed: (): Promise<CodexResetFeed> => invoke("get_codex_reset_feed"),
};

File: src-tauri/src/commands/codex_reset.rs
//! Codex 全局重置预告：读社区 feed（含缓存），供额度面板展示。
//!
//! 按需拉取（与 get_subscription_quota 同款节奏：面板打开/手动刷新时由前端
//! invoke），服务层带 1 小时缓存与旧缓存兜底，见 `services/codex_reset.rs`。

use crate::services::codex_reset;

#[tauri::command]
pub async fn get_codex_reset_feed() -> Result<codex_reset::CodexResetFeed, String> {
    codex_reset::get_feed().await
}
